design/proxy.md

```diff
@@ -0,0 +1,90 @@
+# Proxies
+
+The installer should work behind a proxy.
+
+## Goals
+
+- Proxy installs complete successfully.
+- Docker can pull images from remote and local registries.
+
+## Non Goals
+
+- Automatically detect proxy from environment.
+- Validate kotsadm add-on works with a proxy.
+- Support docker versions below 19.03.
+
+## Background
+
+The kURL spec has a field for an http proxy address, but the feature has not been implemented.
+
+## High-Level Design
+
+Use the proxy when downloading packages from S3.
+Configure docker to pull from remote registries with the proxy and the local registry without the proxy.
+Add `HTTP_PROXY` and `NO_PROXY` environment variables to the kotsadm add-on.
+
+## Detailed Design
+
+### Spec
+
+Proxy configuration uses three fields under the kurl section of the installer spec:
+
+```yaml
+apiVersion: cluster.kurl.sh/v1beta1
+kind: Installer
+metadata:
+  name: proxy
+spec:
+  kurl:
+    HTTPProxy: http://10.128.0.3:3128
+    additionalNoProxy: 10.128.04,10.128.0.5,10.128.0.6
```

Comment:
Should this be array to make it easier to patch?

Reply:
@mzaneri Will array fields work with the spec and CRD?

Reply:
we have 2D arrays for IPTables et al cmds

Reply:
I'm hesitant to change the type on this though, think its a good bit more work for not much gain

Reply:
Also maybe change the name to additionalProxyAddress